src/core/file.c

```diff
@@ -61,10 +62,13 @@ void file_append_extension(char *filename, const char *extension)
     } while (c);
     filename--;
     filename[0] = '.';
-    filename[1] = extension[0];
-    filename[2] = extension[1];
-    filename[3] = extension[2];
-    filename[4] = 0;
+    filename++;
+    int len = strlen(extension);
+    for (int i = 0; i < len; i++) {
+        filename[0] = extension[i];
```

Comment:
In this loop context, I think `*filename` would be a bit less confusing.

Took me a few seconds to guess why it was always assigning to `filename[0]`.

Reply:
Yeah that's definitely better, thanks!